core/invoker/src/main/scala/whisk/core/containerpool/ContainerProxy.scala

```diff
@@ -349,7 +362,15 @@ class ContainerProxy(
     // Only initialize iff we haven't yet warmed the container
     val initialize = stateData match {
       case data: WarmedData => Future.successful(None)
-      case _                => container.initialize(job.action.containerInitializer, actionTimeout).map(Some(_))
+      case _                =>
+        // check if container has generic "prewarm" name
+        if (isPrewarmed(container.name, instance, job.action.exec.kind)) {
+          // now rename container to action based name
+          val newName =
+            ContainerProxy.containerName(instance, job.msg.user.namespace.name.asString, job.action.name.asString)
+          container.rename(newName)
```

Comment:
I think this is asynchronously renaming the container (since `rename` returns a Future, but that Future object is dropped on the floor).  On the one hand, this is good in that it means we are adding minimal latency to an important hot path.  On the other hand, it is not good in that if the rename fails (a) we don't notice and (b) we don't attempt to retry or otherwise handle the error.

Did I read this correctly

Reply:
This is a debug feature for operators? I'm trying to get an idea of the motivating use case here. Users of the platform should not see their containers.

Reply:
Are the suspend and resume operations affected by a pending rename?

Reply:
Good point on possible bug with suspend/resume @rabbah.  That argues that the code should be changed to join the two futures for `rename` and `initialize` before it proceeds with moving to the warm state.   We could still overlap the two operations to attempt to hide the latency of the rename behind the initialize, but we should make sure the rename has completed before continuing.

Reply:
> Are the suspend and resume operations affected by a pending rename?
According to docker daemon source code those operations are synchronized by container object. I don't see there any possible conflict. 
From the OpenWhisk perspective, all container operations currently using container UID.

Reply:
It should be noted that in the past increasing concurrent operations against docker surfaced deadlocks in the daemon.